Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/StatusBean.java

```diff
@@ -52,132 +60,180 @@
+		return certDTOs;
+	}
+
+	public void refresh(boolean refreshMessages) {
+		snapshot = sService.getStatus(refreshMessages);
+		if (refreshMessages) {
+			initPieChart();
+			initBarChart();
+		}
+	}
+
+	public String getOs() {
+		return snapshot.getOs();
+	}
+
+	public String getJava() {
+		return snapshot.getJavaVersion();
+	}
+
+	public String getMemory() {
+		return snapshot.getMemory();
+	}
+
+	public String getAppServer() {
+		return snapshot.getServerVersion();
+	}
+
+	private Map<String, Long> getInboundEventCounts() {
+
+		HashMap<String, Long> map = new HashMap<>();
+		for (EventCount event : snapshot.getEvents().values()) {
+			map.put(event.getEvent(), event.getInbound());
+		}
+		return map;
+	}
+
+	public long getTotalInboundRequests() {
+		long total = 0;
+		for (EventCount event : snapshot.getEvents().values()) {
+			total += event.getInbound();
+		}
+		return total;
+	}
+
+	private Map<String, Long> getOutboundEventCounts() {
+
+		HashMap<String, Long> map = new HashMap<>();
+		for (EventCount event : snapshot.getEvents().values()) {
+			map.put(event.getEvent(), event.getOutbound());
+		}
+		return map;
+	}
+
+	public long getTotalOutboundRequests() {
+		long total = 0;
+		for (EventCount event : snapshot.getEvents().values()) {
+			total += event.getOutbound();
+		}
+		return total;
+	}
+
+	public List<AvailableService> getServices() {
+		return services;
+	}
+
+	public HorizontalBarChartModel getEventBarChart() {
+		return eventBarChart;
+	}
+
+	public PieChartModel getEventPieChart() {
+		if (eventPieChart != null) {
+			refreshPieChart();
+		}
+		return eventPieChart;
+	}
+
+	private void refreshPieChart() {
+
+		for (EventCount event : snapshot.getEvents().values()) {
+			eventPieChart.getData().put(event.getEvent(), event.getTotal());
+		}
+
+		eventPieChart.setFill(false);
+		eventPieChart.setSeriesColors("10253F, CC0000, 33D6FF, FFCC00, 98FB98, FFA500, 00CCFF");
+		eventPieChart.setShowDataLabels(true);
+		eventPieChart.setSliceMargin(5);
+		eventPieChart.setLegendPosition("se");
+	}
+
+	private void initBarChart() {
+		eventBarChart = new HorizontalBarChartModel();
+
+		ChartSeries inboundSeries = createChart("Inbound", getInboundEventCounts());
+		ChartSeries outboundSeries = createChart("Outbound", getOutboundEventCounts());
+
+		eventBarChart.addSeries(inboundSeries);
+		eventBarChart.addSeries(outboundSeries);
+		eventBarChart.setSeriesColors("10253F, CC0000");
+		eventBarChart.setStacked(true);
+		eventBarChart.setLegendPosition("se");
+	}
+
+	private static ChartSeries createChart(String label, Map<String, Long> eventCounts) {
+		ChartSeries series = new ChartSeries();
+		series.setLabel(label);
+		for (Entry<String, Long> serviceEntry : eventCounts.entrySet()) {
+			series.set(serviceEntry.getKey(), serviceEntry.getValue());
+		}
+		return series;
+	}
+
+	private void initPieChart() {
+		eventPieChart = new PieChartModel();
+		refreshPieChart();
+	}
+
+
+	public List<CertificateDTO> getCertificateInfo() {
```

Comment:
This method looks pretty expensive. Lots of "addAll"s. I don't see the need as to why we need expiredCerts and warnningCerts (Misspelled too, by the way). Perhaps consider using an ordered list instead? I would imagine we want to see the certificates with the least amount of valid days left first on the UI anyway - so the order on the list would be the amount of remaining days.

Another potential improvement is simply to iterate over the result of the certificate service directly in a private method which would include lines 218 through 226 here. That would eliminate the need to have to add all of the results (which is already a list) to a bigger list just for the sake of iteration.

Last but certainly not least - certDTO is not cleared when this method is run! You will get duplicated certificates each time this method is run...